src/math/numbers.c

```diff
@@ -177,3 +174,40 @@ uint32_t crc32(uint32_t base, const void *data, uint32_t bytes)
 
 	return ~crc;
 }
+
+/**
+ * Find the last (most-significant) set bit in word x,
+ * https://www.kernel.org/doc/htmldocs/kernel-api/API-fls.html
+ * fls(0) is 0, fls(1) is 1, fls(0x80000000) = 32.
+ */
+int fls(int x)
```

Comment:
I might be wrong, but some arch have machine code for this kind of process, did you check the xtensa ISA?

Reply:
see e.g.
```
src/include/sof/common.h:#define ffs(i) __builtin_ffs(i)
```

Reply:
from Linux kernel:
```
static __always_inline int fls(unsigned int x)
{
	return x ? sizeof(x) * 8 - __builtin_clz(x) : 0;
}
```

Reply:
@andrula-song can you try ```src/include/sof/common.h:#define ffs(i) __builtin_ffs(i)``` as suggested by @lyakh here as this should work with all our compilers.